internal/utils.go

```diff
@@ -0,0 +1,17 @@
+package internal
```

Comment:
that file is a bit weird.
it's called utils but it's in `internal` and it has helper functions.
It'd rather we move the constants that will be reused in a separate package, and rename the package `utils`?

Reply:
My understanding looking at https://github.com/golang-standards/project-layout is that you put in internal things you don't want to expose as public package, so all the utility I think are related to this project but I don't want other people to import.

They do the same here for example: https://github.com/fluxcd/kustomize-controller/tree/main/internal/sops

Do you think we should expose all those utilities?

Reply:
yeah it's more semantics than the layout, I'd name the package utils, in a `utils` directory in the `internal` directory, it would make sense. The import would be `utils`.
I certainly do not think we should expose them.

I'd move the appset constants that will be reused throughout the code base in a separate package. The AppSetKind and APIGroups are not really helpers. This is the last place I would think of to look for this. We could make a new package for it. What do you think? the other two are used locally, happy for them to stay there

Reply:
Cool, I'll do a separate utils package.
Regarding the constant, do you have any suggestion for where to put them?

Reply:
Since `const` is a reserved name, I kept them in utils for the time being, we can move them later.

Reply:
fine, i'll create a quick issue so we don't forget.